File: backend/src/main/java/com/checkmate/backend/domain/store/entity/BusinessHour.java
package com.checkmate.backend.domain.store.entity;

import static jakarta.persistence.ConstraintMode.NO_CONSTRAINT;
import static jakarta.persistence.FetchType.LAZY;
import static jakarta.persistence.GenerationType.IDENTITY;

import com.checkmate.backend.global.entity.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(indexes = @Index(name = "idx_store_id", columnList = "store_id"))
public class BusinessHour extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    @Column(name = "business_hour_id")
    private Long id;

    private String day;
    private String openTime;
    private String closeTime;

    private boolean closed;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "store_id", foreignKey = @ForeignKey(NO_CONSTRAINT))
    private Store store;
}

File: frontend/src/types/shared/businessHour.ts
import type { DayOfWeek } from './dayOfWeek';

export interface BusinessHour {
  dayOfWeek: DayOfWeek;
  openTime?: string; // 00:00 ~ 23:30
  closeTime?: string; // 00:00 ~ 23:30
  is24?: boolean;
  closed?: boolean;
  // client에만 있는 상태, 마감 시간이 다음날 00:00 이후인 경우 체크
  isOver24?: boolean;
}
